apps/projects/app/components/Panel/NewProject/NewProject.js

```diff
@@ -63,160 +206,127 @@ const NewProject = () => {
-            Clear Filters
-          </Text>
-        </Button>
-      </RepoInfo>
-    )
-
-    return (
-      <RepoInfo>
-        <Text>No more repositories to add...</Text>
-      </RepoInfo>
-    )
-  }
-  RepoList.propTypes = {
-    visibleRepos: PropTypes.array.isRequired,
-    repoArray: PropTypes.array.isRequired,
-  }
-
   return (
-    <React.Fragment>
-      <div css={`margin-top: ${3 * GU}px`}>
-        <Text weight="bold">
-            Which repos do you want to add?
-        </Text>
-        <div>
-          <Query
-            fetchPolicy="cache-first"
-            query={GET_REPOSITORIES}
-            onError={console.error}
+    <>
+      {github.status !== 'authenticated' ? (
+        <div style={{ width: '100%', textAlign: 'center', marginBottom: (3 * GU) + 'px' }}>
+          <Button
+            mode="normal"
+            onClick={handleGithubSignIn}
           >
-            {({ data, loading, error, refetch }) => {
-              if (data && data.viewer) {
-
-                const reposDownloaded = filterAlreadyAdded(data.viewer.repositories.edges)
-
-                const visibleRepos = filter ? filterByName(reposDownloaded) : reposDownloaded
-
-
-                const repoArray = visibleRepos.map(repo => ({
-                  title: repo.node.nameWithOwner,
-                  description: '',
-                  node: repo.node,
-                }))
-
-                return (
-                  <div>
-                    <TextInput
-                      type="search"
-                      style={{ margin: '16px 0', flexShrink: '0' }}
-                      placeholder="Search"
-                      wide
-                      value={filter}
-                      onChange={updateFilter}
-                      adornment={
-                        filter === '' && (
-                          <IconSearch
-                            css={`
-                                color: ${theme.surfaceOpened};
-                                margin-right: 8px;
-                              `}
-                          />
-                        )
-                      }
-                      adornmentPosition="end"
-                      ref={searchRef}
-                      aria-label="Search"
-                    />
+            <IconGitHub
+              color={`${theme.surfaceIcon}`}
+              width="18px"
+              height="18px"
+            />
+            <span css="margin-top: 3px; margin-left: 8px">Select from GitHub</span>
+          </Button>
+        </div>
+      ) : (
+        <RepoQuery
+          onRepoSelected={onRepoSelected}
+          repoSelected={repoSelected}
+          setRepoSelected={setRepoSelected}
+        />
+      )}
+      <Button
+        mode="strong"
+        wide
+        onClick={handleNewProject}
+        disabled={repoSelected < 0}
+      >
+        Submit
+      </Button>
+    </>
+  )
+}
+GitHubRepoList.propTypes = PropTypes.func.isRequired
```

Comment:
what's this?

Reply:
It's a shortcut. In case there is only one property one does not need to write:

```
GitHubRepoList.propTypes = {
   handleGithubSignIn: PropTypes.func.isRequired
}
```

it is sufficient to say:

`GitHubRepoList.propTypes = PropTypes.func.isRequired`